main.js

```diff
@@ -0,0 +1,194 @@
+        }
+        game.alter(currentCell.x, currentCell.y, 'Empty')
+        game.alter(nextCell.x, nextCell.y, currentCell.type)
+        if(currentCell.type == 'Player1'){
+            switch(direction){
+                case 'right':
+                    player1.y +=1
+                    break;
+                case 'left':
+                    player1.y -=1
+                    break;
+                case 'up':
+                    player1.x -=1
+                    break;
+                case 'down':
+                    player1.x +=1
+                    break;
+                default:
+            }          
+            player1.coins+= coins
+        }else{
+            switch(direction){
+                case 'right':
+                    player2.y +=1
+                    break;
+                case 'left':
+                    player2.y -=1
+                    break;
+                case 'up':
+                    player2.x -=1
+                    break;
+                case 'down':
+                    player2.x +=1
+                    break;
+                default:
+            } 
+            player2.coins+= coins
+        }
+    }
+}
+
+const moveRight = function(player){
+    const nextCell = {x:player.x, y:player.y+1}
+    moveToCell(player, nextCell, 'right')
+}
+
+const moveLeft = function(player){
+    const nextCell = {x:player.x, y:player.y-1}
+    moveToCell(player, nextCell, 'left')
+}
+
+const moveUp = function(player){
+    const nextCell = {x:player.x-1, y:player.y}
+    moveToCell(player, nextCell, 'up')
+}
+
+const moveDown = function(player){
+    const nextCell = {x:player.x+1, y:player.y}
+    moveToCell(player, nextCell, 'down')
+}
+
+// Up for player 1
+const onPressW = function(){
+    moveUp(player1)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer1(player1.coins)
+}
+// Left for player 1
+const onPressA = function(){
+    moveLeft(player1)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer1(player1.coins)
+}
+// Down for player 1
+const onPressS = function(){
+    moveDown(player1)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer1(player1.coins)
+}
+// Right for player 1
+const onPressD = function(){
+    moveRight(player1)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer1(player1.coins)
+}
+
+// Up for player 2
+const onPressI = function(){
+    moveUp(player2)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer2(player2.coins)
+}
+// Left for player 2
+const onPressJ = function(){
+    moveLeft(player2)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer2(player2.coins)
+}
+// Down for player 2
+const onPressK = function(){
+    moveDown(player1)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer2(player1.coins)
+}
+// Right for player 2
+const onPressL = function(){
+    moveRight(player2)
+    renderer.renderBoard(game.matrix)
+    renderer.renderPlayer2(player2.coins)
+}
```

Comment:
instead of all those `onPress` functions, you could just call the `move` functions from the `start` function and send them the player.... a lot of repetition here

Reply:
I thought it would make the code more readable